README.md

```diff
@@ -196,13 +183,351 @@ Explore the sample by signing in, adding items to the To Do list, Clearing the c
+
+#### Add Todo Item
+
+The method to add a new `Todo` is where we consume our **TodoListService-OBO** Web API, that will consume the downstream **Microsoft Graph** using an access token obtained using the **On-Behalf-Of (OBO) flow**.
+
+To check if the user is signed in, we use the method [GetAccountsAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.identity.client.clientapplicationbase.getaccountsasync?view=azure-dotnet):
+
+```csharp
+var accounts = (await _app.GetAccountsAsync()).ToList();
+if (!accounts.Any())
+{
+   MessageBox.Show("Please sign in first");
+   return;
+}
+```
+Now we call the `AcquireTokenSilent` method to get the cached access token we had obtained earlier during our sign-in. With this token, we can then create a HTTP POST request to our Web API attaching it on the header as `Bearer`.
+
+```csharp
+AuthenticationResult result = null;
+try
+{
+   result = await _app.AcquireTokenSilent(Scopes, accounts.FirstOrDefault())
+      .ExecuteAsync()
+      .ConfigureAwait(false);
+}
+catch (MsalUiRequiredException)
+{
+   MessageBox.Show("Please re-sign");
+}
+catch (MsalException ex)
+{
+   // An unexpected error occurred.
+}
+
+HttpClient httpClient = new HttpClient();
+httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", result.AccessToken);
+
+HttpContent content = new FormUrlEncodedContent(new[] { new KeyValuePair<string, string>("Title", TodoText.Text) });
+
+HttpResponseMessage response = await httpClient.PostAsync(todoListBaseAddress + "/api/todolist", content);
+
+if (response.IsSuccessStatusCode)
+{
+      TodoText.Text = "";
+      GetTodoList();
+}
+else
+{
+      MessageBox.Show("An error occurred : " + response.ReasonPhrase);
+}
+```
+
+Important things to notice:
+
+- After the **SignIn**, the user token will be cached and it can be acquired again by calling [AcquireTokenSilent](https://docs.microsoft.com/en-us/dotnet/api/microsoft.identity.client.iclientapplicationbase.acquiretokensilentasync?view=azure-dotnet).
+- `MsalUiRequiredException` will be thrown if there is no token for that user with the specified scope in the cache, or it got expired. This case requires the user to **SignIn** again.
+- Attaching the access token in the HTTP header as `Bearer` is mandatory when using [**On-Behalf-Of (OBO) flow**](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow).
+
+### TodoListService
+
+The **TodoListService** is our Web API project that will make a call to the downstream **Microsoft Graph API** using an access token obtained via the [**On-Behalf-Of (OBO) flow**](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow). The client that called **TodoListService**, sends a `Bearer` token on the HTTP header and this token will be used to impersonate the user and acquire a proper token for **Microsoft Graph API**.

```

Comment:
>a proper token [](start = 424, length = 14)

another access token

Reply:
typo "access"

---
In reply to: [287246996](https://github.com/Azure-Samples/active-directory-dotnet-webapi-onbehalfof/pull/35#discussion_r287246996) [](ancestors = 287246996)